crates/core/src/operations/mod.rs

```diff
@@ -212,7 +217,13 @@ impl DeltaOps {
     #[cfg(feature = "datafusion")]
     #[must_use]
     pub fn write(self, batches: impl IntoIterator<Item = RecordBatch>) -> WriteBuilder {
-        WriteBuilder::new(self.0.log_store, self.0.state).with_input_batches(batches)
+        let mut wb = WriteBuilder::new(self.0.log_store, self.0.state).with_input_batches(batches);
+        if let Some(parquet_config) = self.0.config.parquet_config {
+            let pc= ParquetConfig::from_str(parquet_config.as_str())
+                .expect("Failed to parse ParquetConfig from table config");
+            wb = wb.with_writer_properties(pc.writer_options());
+        }
+        wb
```

Comment:
All these operations take the associated builder and simply add writer properties. I had written a template to unify this code, but then realized I need to create an actual trait in rust so I abandoned this larger refactor.

Reply:
I think we might want to set the parquet config on the `WriteBuilder` rather than setting the `WriterProperties` here, to allow creating per-file `WriterProperties` later with a KMS and being able to provide the file path and schema. It might get a bit ugly having to deal with backwards compatibility and keeping the option to set the writer properties on the `WriteBuilder` though.

I experimented with a similar idea in my [`encryption_factory` branch](https://github.com/delta-io/delta-rs/compare/main...adamreeve:delta-rs:encryption_factory), which introduces a `WriterPropertiesFactory` type, but that might be replaceable with the `ParquetConfig`.

Reply:
I don't quite follow. What this does is propagate the options set in the `DeltaOp` down to the newly created `WriteBuilder`.

Reply:
Sorry yeah my comment here wasn't very clear, but I think you already had the same idea I was trying to get at in https://github.com/corwinjoy/delta-rs/pull/4#discussion_r2261611847

Basically to support per-file generation of encryption properties we can't create the `WriterProperties` at this point but will need to delay this until a later time when we know the file path and schema.

Maybe that means we pass the `ParquetConfig` directly through to the `WriteBuilder` and eventually the `DeltaWriter`, or maybe we could introduce a factory trait or type to abstract this more.